patches/tModLoader/Terraria.ModLoader/GlobalItem.cs

```diff
@@ -206,12 +206,17 @@ public class GlobalItem
 		/// <summary>
 		/// Allows you to modify the projectile created by a weapon based on the ammo it is using.
 		/// </summary>
+		/// <param name="shooter">The item that is using this ammo</param>
 		/// <param name="item">The ammo item</param>
-		/// <param name="player"></param>
-		/// <param name="type"></param>
-		/// <param name="speed"></param>
-		/// <param name="damage"></param>
-		/// <param name="knockback"></param>
+		/// <param name="player">The player using the item</param>
+		/// <param name="type">The ID of the projectile shot</param>
+		/// <param name="speed">The speed of the projectile shot</param>
+		/// <param name="damage">The damage of the projectile shot</param>
+		/// <param name="knockback">The speed of the projectile shot</param>
+		public virtual void PickAmmo(Item shooter, Item item, Player player, ref int type, ref float speed, ref int damage, ref float knockback) {
```

Comment:
change `item` to `ammo`

Reply:
Done.

Reply:
And then change back `shooter` to `item`? Shooter sounds like the player (the shooter) but is later down the line (Discussion please)

Reply:
Calling it `item` may be confusing since normally `item` would be the item the current GlobalItem instance belongs to, but that would be `ammo` in this case